src/h5web/vis-packs/nexus/NexusPack.test.tsx

```diff
@@ -83,3 +84,76 @@ test('show error when encountering malformed NeXus metadata', async () => {
 
   expect(errorSpy).toHaveBeenCalledTimes(8); // React logs two stack traces per error
 });
+
+test('cancel and retry slow fetch of NxSpectrum', async () => {
+  jest.useFakeTimers();
+  await renderApp();
+
+  await selectExplorerNode('resilience/slow_nx_spectrum');
+  expect(await screen.findByText(/Loading data/)).toBeVisible();
+
+  const errorSpy = mockConsoleMethod('error');
+  userEvent.click(await screen.findByRole('button', { name: /Cancel/ }));
+
+  expect(await screen.findByText('Request cancelled')).toBeVisible();
+  expect(errorSpy).toHaveBeenCalledTimes(2); // React logs two stack traces
+
+  // Move dimension slider
+  const d0Slider = screen.getByRole('slider', { name: 'Dimension slider' });
+  d0Slider.focus();
+  userEvent.keyboard('{PageUp}');
+
+  expect(await screen.findByText('Request cancelled')).toBeVisible();
+  expect(errorSpy).toHaveBeenCalledTimes(4); // React re-logs errors
+  errorSpy.mockRestore();
+
+  userEvent.click(await screen.findByRole('button', { name: /Retry/ }));
+  expect(await screen.findByText(/Loading data/)).toBeVisible();
+
+  // Waterfall!
+  jest.runAllTimers(); // resolve slow fetch right away
+  expect(await screen.findByText(/Loading data/)).toBeVisible();
+  jest.runAllTimers(); // resolve slow fetch right away
+  expect(await screen.findByText(/Loading data/)).toBeVisible();
+  jest.runAllTimers(); // resolve slow fetch right away
+  expect(await screen.findByText(/Loading data/)).toBeVisible();
+  jest.runAllTimers(); // resolve slow fetch right away
+  expect(await screen.findByText(/Loading data/)).toBeVisible();
+  jest.runAllTimers(); // resolve slow fetch right away
+
+  expect(await screen.findByRole('figure')).toBeVisible();
+
+  jest.runOnlyPendingTimers();
+  jest.useRealTimers();
+});
+
+test('cancel and retry slow fetch of NxImage', async () => {
+  jest.useFakeTimers();
+  await renderApp();
+
+  await selectExplorerNode('resilience/slow_nx_image');
+  expect(await screen.findByText(/Loading data/)).toBeVisible();
+
+  const errorSpy = mockConsoleMethod('error');
+  userEvent.click(await screen.findByRole('button', { name: /Cancel/ }));
+  expect(await screen.findByText('Request cancelled')).toBeVisible();
+  expect(errorSpy).toHaveBeenCalledTimes(2); // React logs two stack traces
+  errorSpy.mockRestore();
+
```

Comment:
If we move the slider here, it will request an another slice, right ?

Reply:
- Moving the slider on an `NxSpectrum` visualization now automatically retries fetching the entire dataset, as well as any cancelled supporting datasets (`title`, `axes`, etc.) If some datasets were successfully fetched, they are not refetched.
- Moving the slider on an `NxImage` visualization fetches the new slice and retries fetching any cancelled supporting datasets. Coming back to a cancelled slice automatically retries fetching that slice as well.